src/clr-angular/data/datagrid/_datagrid.clarity.scss

```diff
@@ -774,6 +776,160 @@
     display: block !important;
   }
 
+  /**
+   * Detail pane
+   */
+  .datagrid-outer-wrapper {
+    display: flex;
+    flex-direction: row;
+    flex-grow: 1;
+    overflow: auto;
+  }
+  .datagrid-inner-wrapper {
+    display: flex;
+    flex-direction: column;
+    flex-grow: 1;
+    overflow: auto;
+  }
+  .datagrid-detail-open {
+    .datagrid {
+      border-top-right-radius: 0;
+      border-right: none;
+    }
+    .datagrid-placeholder-container,
+    .datagrid-row {
+      border-right: 1px solid $clr-table-border-color;
+    }
+    .datagrid-footer {
+      border-bottom-right-radius: 0;
+    }
+    .pagination {
+      width: 100%;
+    }
+    .pagination-description-compact {
+      text-align: left;
+      flex: 1;
+    }
+    .datagrid-footer .pagination-list {
+      margin-right: 0;
+    }
+  }
+
+  .datagrid-row-detail-open {
+    position: relative;
+
+    &:before {
+      content: '';
+      display: inline-block;
+      position: absolute;
+      border: 10px solid $clr-table-border-color;
+      border-color: transparent $clr-table-border-color transparent transparent;
+      top: calc(0.5rem - 2px);
+      right: 0;
+    }
+
+    &:after {
+      content: '';
+      display: inline-block;
+      position: absolute;
+      border: 9px solid $clr-datagrid-popover-bg-color;
+      border-color: transparent $clr-datagrid-popover-bg-color transparent transparent;
+      top: calc(0.5rem - 1px);
+      right: -1px;
+    }
+  }
+
+  .datagrid-detail-pane {
+    margin-top: 0.5rem;
+    border: $clr-global-borderwidth solid $clr-datagrid-default-border-color;
+    border-left: none;
+    border-top-right-radius: $clr-global-borderradius;
+    border-bottom-right-radius: $clr-global-borderradius;
+    background: $clr-datagrid-popover-bg-color;
+    overflow: hidden;
+    display: block;
+    flex-grow: 2;
+  }
+
+  .datagrid-detail-pane-content {
+    display: flex;
+    flex-direction: column;
+    overflow: auto;
+    height: 100%;
+  }
+
+  .datagrid-detail-body {
+    flex: 1 1 auto;
+    padding: 0 1rem;
+  }
+
+  .datagrid-detail-header {
+    flex: 0 0 auto;
+    font-size: 0.75rem;
+    line-height: 1.4rem;
+    padding-left: 1rem;
+    margin-top: 0;
+    display: flex;
+    flex-direction: row;
+
+    .datagrid-detail-header-title {
+      flex: 1 1 auto;
+      padding-top: 0.666667rem;
+    }
+
+    .datagrid-detail-pane-close {
+      flex: 1 1 1.5rem;
+      padding: 0;
```

Comment:
Some of the CSS fixes I'd added to help with visual alignment in the header were either not brought over or no longer work as expected in flexbox. Could you address those?